net/dns/manager_test.go

```diff
@@ -598,6 +768,29 @@ func TestManager(t *testing.T) {
 				Routes: upstreams(".", "https://dns.nextdns.io/c3a884"),
 			},
 		},
+		{
+			// on iOS exclusively, tests the split DNS behavior for battery life optimization added in
+			// https://github.com/tailscale/tailscale/pull/10576
+			name: "ios-use-split-dns-when-no-custom-resolvers",
```

Comment:
add the case where it reverts to the previous behavior too (i.e. when there are other split dns routes)? or is that already covered elsewhere, maybe an explicit one is better too.

Reply:
done